This patch fixes (an apparent) bug in Upstart 1.8+ that causes the
undocumented --global extension to the set-env command to fail.
As one of the major reasons for us to move to Upstart 1.8+ was to
get that ability, this fix is needed
--- /init/control.c
+++ /init/control.c
@@ -1202,13 +1202,14 @@
 
 		/* this can be a null value */
 		instance = job_details[1];
-	} else if (getpid () == 1) {
-		nih_dbus_error_raise_printf (
-			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
-			_("Not permissible to modify PID 1 job environment"));
-		return -1;
-	}
-
+	}
+/*  else if (getpid () == 1) {
+ *		nih_dbus_error_raise_printf (
+ *			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
+ *			_("Not permissible to modify PID 1 job environment"));
+ *		return -1;
+ *	}
+ */
 	if (! control_check_permission (message)) {
 		nih_dbus_error_raise_printf (
 			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
@@ -1307,13 +1308,14 @@
 
 		/* this can be a null value */
 		instance = job_details[1];
-	} else if (getpid () == 1) {
-		nih_dbus_error_raise_printf (
-			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
-			_("Not permissible to modify PID 1 job environment"));
-		return -1;
-	}
-
+	}
+/* else if (getpid () == 1) {
+ *		nih_dbus_error_raise_printf (
+ *			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
+ *			_("Not permissible to modify PID 1 job environment"));
+ *		return -1;
+ *	}
+ */
 	/* Verify that job name is valid */
 	if (job_name && ! strlen (job_name)) {
 		nih_dbus_error_raise_printf (DBUS_ERROR_INVALID_ARGS,
@@ -1561,13 +1563,14 @@
 
 		/* this can be a null value */
 		instance = job_details[1];
-	} else if (getpid () == 1) {
-		nih_dbus_error_raise_printf (
-			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
-			_("Not permissible to modify PID 1 job environment"));
-		return -1;
-	}
-
+	}
+/* else if (getpid () == 1) {
+ *		nih_dbus_error_raise_printf (
+ *			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
+ *			_("Not permissible to modify PID 1 job environment"));
+ *		return -1;
+ *	}
+ */
 	if (! control_check_permission (message)) {
 		nih_dbus_error_raise_printf (
 			DBUS_INTERFACE_UPSTART ".Error.PermissionDenied",
